### PR TITLE
fix(ui5-file-uploader): component is successfully rendered on firefox

### PR DESCRIPTION
**Pull request by unazko**
Fixes: #4292

### CODE REVIEW

File: packages/main/src/FileUploader.js [hunk 1]
```diff
@@ -374,7 +374,9 @@ class FileUploader extends UI5Element {
 			formData.append(this.name, this.files[i]);
 		}
 
-		this._internals.setFormValue(formData);
+		if (this._internals.setFormValue) {
+			this._internals.setFormValue(formData);
+		}
```
Comment (dimovpetar) [inline packages/main/src/FileUploader.js:379]: This way the file is not send to the server when the containing form is submitted

**Review by dimovpetar — CHANGES_REQUESTED** [review]
It looks like Firefox added partial support for ElementInternals with version 93 https://developer.mozilla.org/en-US/docs/Web/API/ElementInternals. It only provides access to the shadowRoot, still no native form support.
The problem here is in the getter `_canUseNativeFormSupport`, which gives false positive. My suggestion is to change the code to something like this:
```js
this._internals = this.attachInternals && this.attachInternals();
...

get _canUseNativeFormSupport() {
    return this._internals && this.internals.setFormValue;
}
````

**Review by vladitasev — CHANGES_REQUESTED** [review]
@unazko Please implement Petar's solution, it is the correct one. You cannot just skip `setFormValue` without executing the alternative code.

**Comment by dimovpetar** [comment]
P.S:
Just checked that Firefox has implemented `setFormValue` on 23.10.2021 with this issue https://bugzilla.mozilla.org/show_bug.cgi?id=1556449. It is expected to be part of version 95, but implementing the check I mentioned is the faster solution.

**Comment by unazko** [comment]
Yes, I overlooked that the file won't be sent to the server this way. The proposed solution is now applied.

File: packages/main/src/FileUploader.js [hunk 2]
```diff
@@ -264,7 +264,7 @@ class FileUploader extends UI5Element {
 	constructor() {
 		super();
 		if (this._canUseNativeFormSupport) {
-			this._internals = this.attachInternals();
+			this._internals = this.attachInternals && this.attachInternals();
 		}
```
Comment (dimovpetar) [inline packages/main/src/FileUploader.js:268]: Remove the surrounding if statement. Let it be only:
```js
this._internals = this.attachInternals && this.attachInternals();
```

File: packages/main/src/FileUploader.js [hunk 3]
```diff
@@ -427,7 +427,7 @@ class FileUploader extends UI5Element {
 	}
 
 	get _canUseNativeFormSupport() {
-		return !!this.attachInternals;
+		return this._internals && this.internals.setFormValue;
```
Comment (dimovpetar) [inline packages/main/src/FileUploader.js:430]: `this._internals.setFormValue` (with underscore)

**Review by dimovpetar — CHANGES_REQUESTED** [review]

File: packages/main/src/FileUploader.js (hunk 2 above)
Comment (unazko) [reply packages/main/src/FileUploader.js]: Done

File: packages/main/src/FileUploader.js (hunk 3 above)
Comment (unazko) [reply packages/main/src/FileUploader.js]: Done

**Review by dimovpetar — APPROVED** [review]

**Review by vladitasev — APPROVED** [review]

**Pull request merged**